mojap_metadata/metadata/metadata.py

```diff
@@ -1,22 +1,43 @@
 import json
-from warnings import warn
+import yaml
+from copy import deepcopy
 
 import jsonschema
 
 
 class MetadataProperty:
     def __set_name__(self, owner, name) -> None:
-        self.name = f"_{name}"
+        self.name = name
 
     def __get__(self, obj, type=None) -> object:
-        return getattr(obj, self.name)
+        return obj._data.get(self.name)
 
     def __set__(self, obj, value) -> None:
-        setattr(obj, self.name, value)
+        obj._data[self.name] = value
```

Comment:
Changes to the `MetadataProperty` allows you to persist a schema with extra attributes unused by the `Metadata` class.